code/modules/multiz/structures.dm

```diff
@@ -210,6 +210,10 @@
 				playsound(src, 'sound/effects/stairs_step.ogg', 50)
 				playsound(target, 'sound/effects/stairs_step.ogg', 50)
 
+/obj/structure/stairs/Crossed(obj/O)
+	if(istype(O))
+		O.stair_act()
```

Comment:
I think most of these should only happen if the dir of the object is heading towards the bottom of the stairs, tbh